packages/typescript/src/internal/ts_project.bzl

```diff
@@ -0,0 +1,192 @@
+"ts_project rule"
+
+load("@build_bazel_rules_nodejs//:providers.bzl", "DeclarationInfo", "NpmPackageInfo", "run_node")
+
+_ATTRS = {
+    # TODO: limit to the extensions we expect?
+    # ts, tsx, js, jsx, json
+    "srcs": attr.label_list(allow_files = True),
+    "extends": attr.label_list(allow_files = [".json"]),
+    "tsc": attr.label(default = Label("@npm//typescript/bin:tsc"), executable = True, cfg = "host"),
+    "tsconfig": attr.label(mandatory = True, allow_single_file = [".json"]),
+    "deps": attr.label_list(providers = [DeclarationInfo]),
+}
+
+_OUTPUTS = {
+    "js_outs": attr.output_list(),
```

Comment:
Could you maybe address the difference in outputs of ts_projecrt and ts_library in a comment or readme file somewhere in the PR?

Reply:
Ultimately I think that needs to go in packages/typescript/docs/install.md where there's currently a discussion of `ts_library` vs `@npm//typescript:index.bzl#tsc`

I'll add a comment here about just mirroring what tsc can produce

Reply:
and a TODO about updating the markdown docs after this lands